src/v0/destinations/criteo_audience/networkHandler.js

```diff
@@ -0,0 +1,73 @@
+const {
+  prepareProxyRequest,
+  proxyRequest
+} = require("../../../adapters/network");
+const { isHttpStatusSuccess } = require("../../util/index");
+const {
+  REFRESH_TOKEN
+} = require("../../../adapters/networkhandler/authConstants");
+const tags = require("../../util/tags");
+const {
+  getDynamicErrorType,
+  processAxiosResponse
+} = require("../../../adapters/utils/networkUtils");
+const { NetworkError } = require("../../util/errorTypes");
+
+/**
+ * This function helps to determine type of error occured. According to the response
+ * we set authErrorCategory to take decision if we need to refresh the access_token
+ * or need to disable the destination.
+ * @param {*} code
+ * @param {*} response
+ * @returns
+ */
+//  https://developers.criteo.com/marketing-solutions/v2021.01/docs/how-to-handle-api-errors#:~:text=the%20response%20body.-,401,-Authentication%20error
+const getAuthErrCategory = (status,code) => {
+  if (status === '401' && code === 'authorization-token-invalid') {
+    return REFRESH_TOKEN;
+  }
+
+  return "";
+};
+
+const criteoAudienceRespHandler = (destResponse, stageMsg) => {
+  const { status, response } = destResponse;
+  throw new NetworkError(
```

Comment:
Can we go through their API docs to make sure that we are covering the customer issues ? I mean the customer sent a valid data(according to our source-code) but when the data was sent to criteo APIs we got a `4xx` response back, so we might need to designate it to `NetworkInstrumentationError`. We need to figure out if such scenarios can occur and assign them to the correct error class

Reply:
This proxy is mainly to handle the accessToken expiration. I already got through the possible error codes and 401 unauhtorized seems to be the only one making sense

Reply:
This proxy is mainly to handle the accessToken expiration. I already got through the possible error codes and 401 unauhtorized seems to be the only one making sense

Reply:
It is evident from the changes that the handler handles `accessToken` expiration case, which is good. But what about other possible error-codes that we receive from destination, we should have a way of handling them correctly right ?

Reply:
Yeah I got your point and that is what I mentioned above that I have took a look for the possible error codes (401 and 404). And I am making the needed changes to accomodate 404 along with the test case.
One thing is by default we will be throwing NetworkError so if you are talking about that. Please let me know your Preferable one.

Reply:
What about the validation errors mentioned here ?
https://developers.criteo.com/marketing-solutions/docs/api-error-types#error-category-types
or some of the validation errors mentioned here ?
https://developers.criteo.com/marketing-solutions/docs/audience#audience-validation-errors

Is there no probability of getting these kind of errors at all ?